### PR TITLE
Add options to Makefile builds for deprecated code and warnings

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Comment by crtrott** [comment]
We could consider adding a proper configure option ...

File: Makefile.kokkos [hunk 1]
```diff
@@ -533,6 +533,7 @@ endif
 
 #only add the c++ standard flags if this is not CMake
 tmp := $(call kokkos_append_header,"/* General Settings */")
+tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_DEPRECATED_CODE_3")
```
Comment (dalg24) [inline Makefile.kokkos:536]: How would one enable using the generated makefiles?

**Review by masterleinad — COMMENTED** [review]
IMHO it's OK to enable it unconditionally for Makefile builds (since this is also the default for `CMake` builds).
What's your thought on `KOKKOS_ENABLE_DEPRECATION_WARNINGS`? The default for `CMake` builds would also be to emit deprecation warnings.

**Comment by crtrott** [comment]
OK I added the options to the Makefile now, with deprecated code ON by default and warnings OFF by default.

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

File: Makefile.kokkos [hunk 2]
```diff
@@ -22,7 +22,7 @@ KOKKOS_DEBUG ?= "no"
 KOKKOS_USE_TPLS ?= ""
 # Options: c++14,c++1y,c++17,c++1z,c++2a
 KOKKOS_CXX_STANDARD ?= "c++14"
-# Options: aggressive_vectorization,disable_profiling,enable_large_mem_tests,disable_complex_align
+# Options: aggressive_vectorization,disable_profiling,enable_large_mem_tests,disable_complex_align,disable_deprecated_code,enable_deprecation_warning
```
Comment (dalg24) [inline Makefile.kokkos:25]:
```suggestion
# Options: aggressive_vectorization,disable_profiling,enable_large_mem_tests,disable_complex_align,disable_deprecated_code,enable_deprecation_warnings
```
to be consistent with the CMake options

File: Makefile.kokkos [hunk 3]
```diff
@@ -533,6 +535,12 @@ endif
 
 #only add the c++ standard flags if this is not CMake
 tmp := $(call kokkos_append_header,"/* General Settings */")
+ifneq ($(KOKKOS_INTERNAL_DISABLE_DEPRECATED_CODE), 1)
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_DEPRECATED_CODE_3")
+endif
+ifeq ($(KOKKOS_INTERNAL_ENABLE_DEPRECATION_WARNINGS), 1)
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_DEPRECATION_WRANINGS")
```
Comment (dalg24) [inline Makefile.kokkos:542]:
```suggestion
  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_DEPRECATION_WARNINGS")
```

**Review by dalg24 — CHANGES_REQUESTED** [review]

File: Makefile.kokkos (hunk 2 above)
Comment (crtrott) [reply Makefile.kokkos]: done

File: Makefile.kokkos (hunk 3 above)
Comment (crtrott) [reply Makefile.kokkos]: done

**Review by dalg24 — APPROVED** [review]

**Pull request merged**